sdk/eventhub/event-hubs/src/eventDataBatch.ts

```diff
@@ -114,8 +135,19 @@ export class EventDataBatch {
    * @param eventData  An individual event data object.
    * @returns A boolean value indicating if the event data has been added to the batch or not.
    */
-  public tryAdd(eventData: EventData): boolean {
+  public tryAdd(eventData: EventData, options: TryAddOptions = {}): boolean {
     throwTypeErrorIfParameterMissing(this._context.connectionId, "eventData", eventData);
+
+    let isInstrumented = false;
+    // check if the event has already been instrumented
+    if (!eventData.properties || !eventData.properties[TRACEPARENT_PROPERTY]) {
+      const messageSpan = createMessageSpan(options.parentSpan);
+      // Create a shallow copy of eventData and eventData.properties in case we add the diagnostic id to the properties.
+      eventData = {...eventData, properties: {...eventData.properties}};
+      isInstrumented = instrumentEventData(eventData, messageSpan);
```

Comment:
If we didn't want instrumentEventData to mutate the object, we could just treat its input as readonly and do the object copy trick inside it and return the new version, right?

It just feels odd to have a method that mutates a passed-in argument if the only user of that method doesn't want that behavior

Reply:
Yeah, I originally had that but then came across a situation where I also needed to know if the event was instrumented by the method. I didn't want to make the return type return the boolean and the event data, but maybe I can tackle the `isInstrumented` check another way.

Reply:
Ah, I remember now. I wanted to know if an event being added to a batch via `tryAdd` was instrumented by `instrumentEventData`, or if it was already instrumented before being added. I can be smarter about handling this in `tryAdd` and have `instrumentEventData` go back to returning a new copy of EventData.